### ISSUE TITLE
New list cannot be deleted by right click

### ISSUE THREAD

**Issue by jialiren** [LLK/scratch-gui#3007]
New list can not be deleted by right mouse button, right key can be deleted in scratch2.0

**Comment by Wgil** [comment]
Hello,

How can I reproduce it?

**Comment by alexmotoc** [comment]
Go to Variables and create a new list. Try right clicking on the list you've just created and there is no menu to delete present. This functionality is available in 2.0 and also for normal variables.

Expected behaviour:
<img width="317" alt="screen shot 2018-09-07 at 14 22 26" src="https://user-images.githubusercontent.com/16880268/45216605-d7878480-b2a9-11e8-9319-37babb81249f.png">

Am I correct? I could try to look into this - I am a beginner with React and new to this codebase but feel it will be a good way to make my first open source contribution!

**Comment by joker314** [comment]
Yeah, that's right!

<hr>

Should we also add a 'rename list' option (this is _not_ available in 2.0 for lists, but is for variables)

**Comment by towerofnix** [comment]
@alexmotoc Unfortunately, to make scratch-gui see your changes, you need to re-build the entire scratch-blocks (in my experience), by doing `npm run prepublish`. (And then you have to restart the gui server, since it won't automatically detect that scratch-blocks was updated.)

Do note that you can test your scratch-blocks changes simply by opening the `tests/vertical_playground.html` file in your web browser, though. That will automatically adjust to changes you make in the source code. You don't even have to rebuild scratch-blocks, just reload the webpage.

Oh, by the way, my steps for linking scratch-blocks are "manual" - I don't use `npm link`:

```bash
scratch-gui $ npm install
scratch-gui $ rm -rf node_modules/scratch-blocks
scratch-gui $ ln -s ~/Documents/scratch-blocks node_modules/scratch-blocks
# You don't need to do anything in scratch-blocks with this method of linking
# (besides rebuilding with npm run prepublish, of course).
```

**Comment by alexmotoc** [comment]
@towerofnix Thanks a lot! I ended up testing the changes using the `tests/vertical_playground.html` file and it was sufficient for my needs.

I opened a pull request in the other repo - https://github.com/LLK/scratch-blocks/pull/1715. Will wait for a review. I am not too happy about almost duplicating the other context menu. Was wondering if there was a good way to refactor that and keep it in one function since they are basically identical.
